Dockerfile

```diff
@@ -0,0 +1,74 @@
+#
+# Image with pre-built tools
+#
+FROM zenithdb/compute-tools:latest AS compute-deps
+# Only to get ready apply_conf binary as a dep
+
+#
+# Image with Postgres build deps
+#
+FROM debian:buster-slim AS build-deps
+
+RUN apt-get update && apt-get -yq install automake libtool build-essential bison flex libreadline-dev zlib1g-dev libxml2-dev \
+                                          libcurl4-openssl-dev
+
+#
+# Image with built Postgres
+#
+FROM build-deps AS pg-build
+
+# Add user postgres
+RUN adduser postgres
+RUN mkdir /pg && chown postgres:postgres /pg
+
+# Copy source files
+COPY . /pg/
+
+# Build and install Postgres locally
+RUN mkdir /pg/compute_build && cd /pg/compute_build && \
+    ../configure CFLAGS='-O0 -g3' --prefix=$(pwd)/postgres_bin --enable-debug --enable-cassert --enable-depend && \
+    # Install main binaries and contribs
+    make MAKELEVEL=0 -j $(getconf _NPROCESSORS_ONLN) -s install && \
```

Comment:
What's the benefit of `_NPROCESSORS_ONLN` over `nproc`?

Reply:
There is no `nproc` on macOS and I just use `getconf _NPROCESSORS_ONLN` everywhere. Here is a linux container, so it isn't needed, but I do copy-paste some script parts to/from local scripts and this is just more convinient